data/abi/abi_encode_test.go

```diff
@@ -34,6 +35,9 @@ func TestEncodeValid(t *testing.T) {
 	// randomly pick 1000 valid uint values and check if encoded value match with expected
 	for intSize := 8; intSize <= 512; intSize += 8 {
 		upperLimit := big.NewInt(0).Lsh(big.NewInt(1), uint(intSize))
+		uintType, err := MakeUintType(uint16(intSize))
```

Comment:
Why does MakeUintType take a uint16?  I'd imagine you'd be better off with int.  You'll still need bounds checking, but it will avoid lots of casts when calling it.